src/main/java/org/myrobotlab/swing/ProgramABGui.java

```diff
@@ -140,36 +143,23 @@ public void actionPerformed(ActionEvent event) {
       if (filter.isSelected()) {
         textFiltered = textFiltered.replace("'", " ").replace("-", " ");
       }
-      Response answer = (Response) swingGui.sendBlocking(boundServiceName, 10000, "getResponse", textFiltered);
       // response.setText(response.getText() + "<br/>\n\r" + answer);
-      if (answer != null) {
-        response.append("\n" + answer.msg.trim());
-      } else {
-        response.append("\nERROR: NULL Response");
-      }
+      send("getResponse", textFiltered);
+
       // clear out the original question.
       text.setText("");
       response.setCaretPosition(response.getDocument().getLength());
     } else if (o == startSessionButton) {
-      String path = progABPath.getText().trim();
-      String user = userName.getText().trim();
-      String bot = botName.getText().trim();
-      swingGui.send(boundServiceName, "setPath", path);
-      swingGui.send(boundServiceName, "reloadSession", path, user, bot);
-
-    } else if (o == killAiml) {
       String path = progABPath.getText().trim();
       String user = userName.getText().trim();
       String bot = botName.getText().trim();
       swingGui.send(boundServiceName, "setPath", path);
       swingGui.send(boundServiceName, "reloadSession", path, user, bot, true);
     } else if (o == reloadSession) {
-      swingGui.send(boundServiceName, "setUsername", userName.getText().trim());
+      swingGui.send(boundServiceName, "startSession", userName.getText().trim());

```

Comment:
Start session doesn't actually reload the whole chat bot if the chat bot is already created.  Reload Session should reload the AIML and the entire chat bot brain.

Reply:
you right, button identification problem. tks for pointed it